src/classes/GiftBatches.cls

```diff
@@ -0,0 +1,55 @@
+/*
+    Copyright (c) 2021 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @description Domain class for Gift Batches in NPSP.
+*/
+public inherited sharing class GiftBatches extends fflib_SObjects {
+
+    public GiftBatches(Set<Id> ids, GiftBatchSelector selector) {
+        this(selector.selectGiftBatchesById(ids));
+    }
+
+    public GiftBatches(List<DataImportBatch__c> records) {
+        super(records, DataImportBatch__c.SObjectType);
+    }
+
+    public Map<Id, Map<String, Integer>> getTotalsMap() {
+        Map<Id, Map<String, Integer>> totalsByBatchId = new Map<Id, Map<String, Integer>>();
+
+        for (SObject batch : getRecords()) {
+            List<AggregateResultProxy> totals = new GiftBatchSelector().selectBatchGiftsGroupedByStatusBy(batch.Id);
```

Comment:
We might hit some governor limits here, since `new GiftBatchSelector().selectBatchGiftsGroupedByStatusBy(batch.Id);` is essentially a query. We might have to bulkify this call instead

Reply:
Ah totally right! I'll make that update

Reply:
Updated!